### ISSUE TITLE
Join pattern

### ISSUE THREAD

**Issue by iluwatar** [iluwatar/java-design-patterns#70]
**Description:**
The Join design pattern allows multiple concurrent processes or threads to be synchronized such that they all must complete before any subsequent tasks can proceed. This pattern is particularly useful in scenarios where tasks can be executed in parallel but the subsequent tasks must wait for the completion of these parallel tasks.

**Main Elements of Join Pattern:**
1. **Task Execution:** Multiple tasks are executed concurrently.
2. **Synchronization Point:** A join point where all tasks must converge and synchronize.
3. **Completion Handling:** After all tasks reach the join point, subsequent tasks can proceed.

**References:**
- [Join pattern on Wikipedia](https://en.wikipedia.org/wiki/Join-pattern)

**Acceptance Criteria:**
1. Implement a `JoinPattern` class demonstrating the synchronization of multiple threads using the join pattern.
2. Include a comprehensive unit test that validates the correct synchronization and execution order of threads.
3. Ensure the implementation follows the project contribution guidelines as outlined in the [wiki](https://github.com/iluwatar/java-design-patterns/wiki).

**Comment by AlesandraSzumacher** [comment]
@iluwatar I with @marianka600 want to work on this pattern. Can we?

**Comment by iluwatar** [comment]
Great, please go ahead @AlesandraSzumacher and @marianka600

**Comment by iluwatar** [comment]
@AlesandraSzumacher and @marianka600  are you still implementing this?

**Comment by iluwatar** [comment]
This issue is free again

**Comment by iluwatar** [comment]
Updated task description

**Comment by keshavMM004** [comment]
Hey I am a lot excited for this problem .I have good hand on java multithreading , syncrhronization , locks . I want to work on this  . Can I ?

**Comment by nour-awad** [comment]
hello @iluwatar, can I work on this issue

**Comment by iluwatar** [comment]
@keshavMM004 are you still working on it?

**Comment by keshavMM004** [comment]
Sorry I missed the update on this . I will submit it within a week .

**Comment by keshavMM004** [comment]
> @keshavMM004 are you still working on it?

Hey facing difficulties with pom.xml and Logger and checkstyle violations . 
Please help

**Comment by iluwatar** [comment]
Duplicate of https://github.com/iluwatar/java-design-patterns/issues/3227
